### PR TITLE
Feature/44authorisation

### PR DESCRIPTION
**Pull request by te4-david-petersson**
#44 
Login isn't actually a thing yet, but need views to "redirect" to logged in pages

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'sinatra'
+require_relative 'deps.rb'
```
Comment (te4-gabriel-stark) [inline app.rb]: You don't need .rb if it's require_relative.

File: deps.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require_relative 'models/student.rb'
+require_relative 'models/user.rb'
+require_relative 'models/teacher.rb'
+require_relative 'controllers/userController.rb'
+
```
Comment (te4-gabriel-stark) [inline deps.rb]: .rb

File: models/user.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        @password
+    end
+
+    def github_name
+        @github_name
+    end
+
+    def teacher
+        @teacher
+    end
```
Comment (te4-gabriel-stark) [inline models/user.rb]: Couldn't you use a attr_reader for this instead?

File: controllers/userController.rb [hunk 4]
```diff
@@ -12,15 +12,27 @@ def register(params)
...
+    
     name = params["name"]
     password = params["password"]
     user = User.getName(name)
-    hashed_password = BCrypt::Password.new(user.password)
-    id = user.id
-    if hashed_password == password
+    if user != nil
+        hashed_password = BCrypt::Password.new(user.password)
+        id = user.id
```
Comment (TE4-emanuel-strom) [inline controllers/userController.rb]: What is the perpus of this variable?

**Pull request merged**